src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -76,7 +76,11 @@ const useFetchChannelHook = () => {
 
     try {
       const channelList = ChannelList.fromChannelAPI(channel, type[channelType]);
-      await channelList.saveIfLatest(localDb);
+      if (channel?.type === 'topics') {
+        await channelList.save(localDb);
+      } else {
+        await channelList.saveIfLatest(localDb);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new logic introduced here checks the channel type and decides whether to save the channel list directly or only if it's the latest. This is a good approach for handling different types of channels, but there's a potential issue: what if `channel` is null or undefined? The code will throw an error when trying to access the `type` property. To prevent this, you should add a null check before accessing the `type` property.
```diff
-      if (channel?.type === 'topics') {
+      if (channel && channel.type === 'topics') {
```
Also, consider adding more specific error handling rather than just logging a generic message. It would be beneficial to log the actual error message to help with debugging.
```diff
-      console.log('error on helperChannelPromiseBuilder');
+      console.log('error on helperChannelPromiseBuilder:', e.message);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->